### PR TITLE
Fixes #25165 - Add Registered hosts report

### PR DESCRIPTION
**Pull request by mbacovsky**
PR Katello/katello#7747 is required for this report.
I'd like to ask reviewer to check if the queries could be further optimized and if the data collected match what was requested.

### CODE REVIEW

File: report_templates/registered_hosts.erb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  required: true
+  input_type: user
+  description: Format of the generated report (csv or yaml)
+  options: "csv\r\nyaml"
+  advanced: false
+model: ReportTemplate
+require:
+- plugin: katello
+￼ version: 3.9.0
+%>
```
Comment (ares) [inline report_templates/registered_hosts.erb]: missing `-`

File: report_templates/registered_hosts.erb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+name: Registered hosts
+snippet: false
+template_inputs:
+- name: Hosts filter
+  required: false
+  input_type: user
+  description: Limit the report only on hosts found by this search query. Keep empty
+    for report on all available hosts.
+  advanced: false
+- name: Output format
```
Comment (ares) [inline report_templates/registered_hosts.erb]: as in other PR, I don't think this should be exposed as input as of now

**Review by ares — CHANGES_REQUESTED** [review]
the similar feedback as elsewhere, otherwise seems to work fine

**Comment by mbacovsky** [comment]
@ares, updated.

File: report_templates/registered_hosts.erb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  input_type: user
+  description: Limit the report only on hosts found by this search query. Keep empty
+    for report on all available hosts.
+  advanced: false
+model: ReportTemplate
+require:
+- plugin: katello
+￼ version: 3.9.0
+-%>
+<%- load_hosts(search: input('Hosts filter'), includes: [:operatingsystem,:subscriptions,:fact_values,:fact_names,:applicable_errata,:applicable_rpms]).each_record do |host| -%>
```
Comment (ares) [inline report_templates/registered_hosts.erb]: sorry, I missed that in first review, could you please add spaces after commas here?

**Comment by mbacovsky** [comment]
@ares fixed, please squash it during the merge

**Review by ares — APPROVED** [review]

**Comment by ares** [comment]
Thanks @mbacovsky, merging

**Pull request merged**